codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoSettings.java

```diff
@@ -33,10 +34,16 @@ public final class GoSettings {
     private static final String SERVICE = "service";
     private static final String MODULE_NAME = "module";
     private static final String MODULE_DESCRIPTION = "moduleDescription";
+    private static final String MODULE_VERSION = "moduleVersion";
+    private static final String GENERATE_GO_MOD = "generateGoMod";
+    private static final String GO_DIRECTIVE = "goDirective";
 
     private ShapeId service;
     private String moduleName;
     private String moduleDescription = "";
+    private String moduleVersion;
+    private Boolean generateGoMod = false;
+    private String goDirective = "1.15";
```

Comment:
Could the values for these be extracted into static fields, e.g. for `goDirective`:

```java
private static final String DEFAULT_GO_DIRECTIVE = "1.15";
```